app/main/lib/image_similarity.py

```diff
@@ -60,47 +61,54 @@ def add_image(save_params):
     db.session.rollback()
     raise e
 
-def search_image(params):
-  url = params.get("url")
-  context = params.get("context")
-  threshold = params.get("threshold")
-  limit = params.get("limit")
-  if not context:
-    context = {}
-  if not threshold:
-    threshold = 0.9
-  if url:
-    image = ImageModel.from_url(url, None)
-    model=app.config['IMAGE_MODEL']
-    if model and model.lower()=="pdq":
-      app.logger.info(f"Searching with PDQ.")
-      result = search_by_pdq(image.pdq, threshold, context, limit)
+def callback_add(task):
+    return media_crud.add(task, ImageModel, ["pdq", "phash"])
+
+def blocking_search_image(task):
+    image, temporary, context, presto_result = media_crud.get_blocked_presto_response(task, ImageModel, "image")
+    threshold = task.get("threshold", 0.0)
+    limit = task.get("limit", 200)
+    model = app.config['IMAGE_MODEL']
+    if image:
+        if model and model.lower() == "pdq":
+            app.logger.info(f"Searching with PDQ.")
+            image.pdq = presto_result["body"]["hash_value"]
+            result = search_by_pdq(image.pdq, threshold, context[0], limit)
+        else:
+            app.logger.info(f"Searching with phash.")
+            image.phash = presto_result["body"]["hash_value"]
+            result = search_by_phash(image.phash, threshold, context[0], limit)
+        if temporary:
+            media_crud.delete(task, ImageModel)
+        else:
+            media_crud.save(image, ImageModel, ["pdq", "phash"])
+        if task.get("limit"):
+            return {"result": result[:task.get("limit")]}
+        else:
+            return {"result": result}
     else:
-      app.logger.info(f"Searching with phash.")
-      result = search_by_phash(image.phash, threshold, context, limit)
-  else:
-    result = search_by_context(context, limit)
-  return {
-    'result': result
-  }
+        return {"error": "Image not found for provided task", "task": task}
+
+def async_search_image(task, modality):
+    return media_crud.get_async_presto_response(task, ImageModel, modality)
 
 @tenacity.retry(wait=tenacity.wait_fixed(0.5), stop=tenacity.stop_after_delay(5), after=_after_log)
 def search_by_context(context, limit=None):
   try:
     context_query, context_hash = get_context_query(context)
     if context_query:
       cmd = """
-          SELECT id, sha256, phash, url, context FROM images
+          SELECT id, doc_id, phash, url, context FROM images
           WHERE 
         """+context_query
     else:
       cmd = """
-          SELECT id, sha256, phash, url, context FROM images
+          SELECT id, doc_id, phash, url, context FROM images
```

Comment:
has the table structure changed? are we now using doc_id instead of hash to look up images?  Is this compatible with data already in the db?

Reply:
Seconding this... is it a simple change in the column name or is it a different value being stored? Is it going to impact existing data, or new images will still be able to match existing images?

Reply:
This is not going to break anything downstream - we just don't need to return the sha because no one uses it, while the doc_id is much more useful